bundles/org.openhab.binding.surepetcare/src/main/java/org/openhab/binding/surepetcare/internal/utils/ByteArrayFileCache.java

```diff
@@ -0,0 +1,306 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.surepetcare.internal.utils;
+
+import java.io.File;
+import java.io.IOException;
+import java.math.BigInteger;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.config.core.ConfigConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is a simple file based cache implementation.
+ *
+ * @author Christoph Weitkamp - Initial contribution
```

Comment:
Did @cweitkamp contribute it here?

Reply:
I added this for the pet picture cache... and this file was taken from the OWM binding... so i left it as is to credit the author

Reply:
@cweitkamp what do you think how should we handle this?

Reply:
I am fine with it. But we have to add myself with an "Also-By" line to the commit message.

FTR: This is the fifth binding which makes use of this util class (see https://github.com/openhab/openhab-addons/issues/4736). You can find the most recent version of it in the Chromecast binding - introduced in https://github.com/openhab/openhab-addons/pull/7367). You may want to pick that one.

I will add this binding to the list in #4736. I once tried to move it over to OHC but it did not work nicely. My current implementation uses `ConfigConstants.getUserDataFolder()` which is not available in package `org.openhab.core.cache`. I have to solve one of the todos first. There are even bigger plans for it (see e.g. eclipse-archived/smarthome#6041) to make it configurable and it could be used for voice add-ons to cache similar speech data.

Reply:
Thanks.
i will update this file with the newer one.

Reply:
Hey @cweitkamp regarding the "Also-by" that we need to the commit message... 

Can you please check on this commit if i've done it right... Thanks

https://github.com/HerzScheisse/openhab-addons/commit/a46ce12d41a5bad5c88c72e306b8dbf8d0dbbffd

Reply:
With removal of this file you can change the sign-off again. It will not be necessary to mention me in this PR 😉 .

Reply:
this uses now the common class in the core